### DISCUSSION TITLE
Automatic system for detection and measurement of pipe openings

### DISCUSSION THREAD

**Discussion by MuhammedSYeter** [community/community#120934] [Programming Help]
### Body

I want to use 3D scanner to measure pipe openings in the production line. Therefore, I need an automated software system for that operation. Can you guys guide me in this problem. 

System will take the the 3D point cloud from the 3D scanner and automatically use this point cloud to detect pipe openings and measure their inner diameter, outer diameter, wall thickness and ovality.

Thanks in advance

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ebndev** [comment]
Hey @MuhammedSYeter!

Were you able to find resources on this? It doesn't look like we have that many posts in the Community about 3D printing but it would be great to reference this post in the future if someone else encounters an issue.
